### PR TITLE
update overseas telephone link

### PR DESCRIPTION
**Pull request by goodviber**
### JIRA ticket number
https://dfedigital.atlassian.net/browse/GITPB-596
### Context
The link to Telephone on equivalent/overseas_completion page was displaying as "not visited"
### Changes proposed in this pull request
updated telephone partial to provide correct equivalent link

### CODE REVIEW

File: app/views/check_answers/_telephone.html.erb [hunk 1]
```diff
@@ -7,8 +7,12 @@
   </dd>
   <dd class="govuk-summary-list__actions">
     <% if defined?(link) %>
```
Comment (ethax-ross) [inline app/views/check_answers/_telephone.html.erb:9]: This section could be simplified to:

```
<% if defined?(link) && link == Equivalent::LINK %>
        <%= show_link("#{link}" + 'overseas_candidate') %>
<% else %>
        <%= show_link("#{link}" + 'overseas_telephone') %>
<% end %>
```

Also, one for another PR but it's not very clear what step `overseas_candidate` is - you have to dig up to the parent model:

```
OverseasCandidate < Callback
```

^ this inheritance would be clearer if the model was something like `OverseasCallback` or `OverseasCandidateCallback`

Comment (goodviber) [reply app/views/check_answers/_telephone.html.erb]: Not sure that works, the else clause at line 15 needs to cover the 'no link' case ?

Comment (goodviber) [reply app/views/check_answers/_telephone.html.erb]: image

Comment (ethax-ross) [reply app/views/check_answers/_telephone.html.erb]: Ah you're right, I misread that - thought lines 12 and 15 were the same

**Pull request merged**